### DISCUSSION TITLE
Error after execute

### DISCUSSION THREAD

**Discussion by popoveo** [LTRData/FuseDotNet#1] [General]
Writing "Cannot load DLL "fuse3"...

**Comment by Olof-Lagerkvist** [comment]
You need to install fuse3 on your system for this to work. What operating system are you using?

**Reply by popoveo** [reply]
Windows 11. How can I install fuse3?
Thanks.

**Reply by Olof-Lagerkvist** [reply]
No, sorry, this will not work on Windows. Maybe you are looking for Dokan? I have a fork of dokan-dotnet library if you want to build virtual file systems on Windows.

**Reply by popoveo** [reply]
I worked with Dokan - everything is fine there. It was interesting to try with this library.
In principle, I found the source code of fuse 3 for Widows in the WinFst project, but a wrapper for C# is needed.

**Reply by Olof-Lagerkvist** [reply]
Okay. I have only tested this library on Linux and FreeBSD so far. It might be that you could make it work on Windows too, but I am a bit unsure. You would need a DLL that works like libfuse3.so on *nix platforms. Also, the structures sent between kernel and user mode needs to have a layout exactly matched by .NET structs in this library.

I would definitely recommend Dokan instead when you implement file systems in user mode in Windows.

**Comment by popoveo** [comment]
Thank you, if something works out for Windows, I'll write in the thread

**Reply by Olof-Lagerkvist** [reply]
Thank you!
